wallet-new/server/Cardano/Wallet/API/V1/Handlers/Accounts.hs

```diff
@@ -41,9 +40,9 @@ getAccount wId accId =
 listAccounts
     :: (MonadThrow m, V0.MonadWalletLogicRead ctx m)
     => WalletId -> RequestParams -> m (WalletResponse [Account])
-listAccounts wId params =
-    let accounts = migrate wId >>= V0.getAccounts . Just >>= migrate @[V0.CAccount] @[Account]
-    in respondWith params (const accounts)
+listAccounts wId params = do
+    accounts :: [Account] <- migrate wId >>= V0.getAccounts . Just >>= migrate
```

Comment:
@sectore Probably it doesn't matter too much now, but note how the two are not the same: the previous implementation is lazy and will defer running the monadic action up until necessary, whereas this one runs it immediately. Any chance we can restore the old behaviour?

Reply:
@adinapoli-iohk As mentioned in Slack previous implementation did throw following error with latest `feature/new-wallet`.
```
cardano-sl/wallet-new/server/Cardano/Wallet/API/V1/Handlers/Accounts.hs:45:62: error:
    • Couldn't match type 'V0.CAccount' with 'Account'
      Expected type: [V0.CAccount] -> [] V0.CAccount (f0 Account)
        Actual type: [Account] -> [] V0.CAccount (f0 Account)
    • In the second argument of '(>>=)', namely
        'migrate @[V0.CAccount] @[Account]'
      In the expression:
        migrate wId >>= V0.getAccounts . Just
        >>= migrate @[V0.CAccount] @[Account]
      In an equation for 'accounts':
          accounts
            = migrate wId >>= V0.getAccounts . Just
              >>= migrate @[V0.CAccount] @[Account]

/cardano-sl/wallet-new/server/Cardano/Wallet/API/V1/Handlers/Accounts.hs:45:71: error:
    • Expecting one fewer argument to '[V0.CAccount]'
      Expected kind '* -> *', but '[V0.CAccount]' has kind '*'
    • In the type '[V0.CAccount]'
      In the second argument of '(>>=)', namely
        'migrate @[V0.CAccount] @[Account]'
      In the expression:
        migrate wId >>= V0.getAccounts . Just
        >>= migrate @[V0.CAccount] @[Account]
```
I can't say exactly why this happened now, because everything seems to work before. A very similar issue seems to be fixed w/ f71dc80dd7317e15519e04923fcaf8b2a1355e17 in a same way.

Reply:
@adinapoli-iohk @sectore yes I run into the same problem and fixed it with https://github.com/input-output-hk/cardano-sl/commit/f71dc80dd7317e15519e04923fcaf8b2a1355e17 . Didn't waste too much time to bother with laziness though.